indico/modules/events/registration/models/registrations.py

```diff
@@ -697,3 +731,48 @@ def _set_checked_in_dt(target, value, *unused):
     @listens_for(Registration.transaction, 'set')
     def _set_transaction_id(target, value, *unused):
         value.registration = target
+
+
+class RegistrationTag(db.Model):
+    """Registration tag used to mark/filter registrations."""
+    __tablename__ = 'registration_tags'
+    __table_args__ = {'schema': 'event_registration'}
```

Comment:
I'd add a check constraint that ensures tag names are unique. See `ix_uq_tags_event_id_code_lower` for an example in Editing tags